power_grid_model_c/power_grid_model/include/power_grid_model/optimizer/tap_position_optimizer.hpp

```diff
@@ -33,13 +42,63 @@ inline auto build_transformer_graph(State const& /*state*/) -> TransformerGraph
     return {};
 }
 
-inline auto get_edge_weights(TransformerGraph const& /*graph*/) -> std::vector<std::pair<Idx2D, EdgeWeight>> {
-    // TODO(mgovers): implement
-    return {};
+inline auto get_edge_weights(TransformerGraph const& graph) -> std::vector<std::pair<Idx2D, EdgeWeight>> {
+    // Step 2: Initialize the rank of all vertices (transformer nodes) as infinite (INT_MAX)
+    std::vector<EdgeWeight> rank(boost::num_vertices(graph), infty);
+    std::vector<Idx2D> sources(boost::num_vertices(graph));
+
+    // Step 3: Loop all the connected sources (status == 1)
+    for (auto v : boost::make_iterator_range(boost::vertices(graph))) {
+        if (graph[v].status == 1) {
```

Comment:
magic value. you can declare `constexpr Idx source_vertex = 1` and `constexpr Idx non_source_vertex = 0` earlier in the code (either outside where you initially declared the `status` or within this function)